src/c2goto/library/pthread_lib.c

```diff
@@ -691,3 +691,19 @@ __ESBMC_HIDE:;
   __ESBMC_atomic_end();
   return result;
 }
+
+
+int pthread_equal(pthread_t tid1, pthread_t tid2)
+
+{
+    __ESBMC_HIDE:;
+    __ESBMC_atomic_begin();
+     tid1 = __ESBMC_get_thread_id();
+     tid2 = __ESBMC_get_thread_id();
+    if (tid1 == tid2)
+        return 1;
+    else
+        return 0;
+    __ESBMC_atomic_end();
+}
```

Comment:
This will cause problems because you are returning before closing the atomic scope. The code should be:
```
  __ESBMC_HIDE:;
  __ESBMC_atomic_begin();
  tid1 = __ESBMC_get_thread_id();
  tid2 = __ESBMC_get_thread_id();
  bool res = tid1 == tid2;
  __ESBMC_atomic_end();
  return res;
```

Reply:
Why are we using `__ESBMC_get_thread_id();`?

The ID of the threads are given as argument in the function.

Shall it be written as?

```
  __ESBMC_HIDE:;
  __ESBMC_atomic_begin();
  return tid1 == tid2;
  __ESBMC_atomic_end();
```